qa/tasks/cephfs/test_volume_client.py

```diff
@@ -46,7 +46,7 @@ def _volume_client_python(self, client, script, vol_prefix=None, ns_prefix=None)
 vc.disconnect()
         """.format(payload=script, conf_path=client.config_path,
                    vol_prefix=vol_prefix, ns_prefix=ns_prefix),
-        self.py_version)
+        'python3')
```

Comment:
i think we should continue using `self.py_version`.

Reply:
Local test, the python2 will fail: 

from ceph_volume_client import CephFSVolumeClient, VolumePath

It says couldn't find the ceph_volume_client .

Reply:
With this PR fix there still has the` AttributeError: 'TestVolumeClient' object has no attribute 'assertCountEqual'` error.

```
2020-04-18T01:50:21.920 INFO:tasks.cephfs_test_runner:======================================================================
2020-04-18T01:50:21.920 INFO:tasks.cephfs_test_runner:ERROR: test_get_authorized_ids (tasks.cephfs.test_volume_client.TestVolumeClient)
2020-04-18T01:50:21.920 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2020-04-18T01:50:21.921 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2020-04-18T01:50:21.921 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-lxb-volume-client-py3/qa/tasks/cephfs/test_volume_client.py", line 659, in test_get_authorized_ids
2020-04-18T01:50:21.921 INFO:tasks.cephfs_test_runner:    self.assertCountEqual(str(expected_result), auths)
2020-04-18T01:50:21.921 INFO:tasks.cephfs_test_runner:AttributeError: 'TestVolumeClient' object has no attribute 'assertCountEqual'
2020-04-18T01:50:21.922 INFO:tasks.cephfs_test_runner:

```

Reply:
Tried get the version by using `sys.version_info.major`, but we get Python2, while the teuthology test suite passed config with Python3.

```
2020-04-19T09:48:22.060 INFO:teuthology.orchestra.run:Running command with timeout 900
2020-04-19T09:48:22.060 INFO:teuthology.orchestra.run.smithi093:> (cd /home/ubuntu/cephtest && exec sudo chmod 1777 /home/ubuntu/cephtest/mnt.3)
2020-04-19T09:48:22.144 INFO:tasks.cephfs.test_volume_client:using python version: python3
2020-04-19T09:48:22.145 INFO:tasks.cephfs.test_volume_client:from sys.version_info.major --> using python version: 2
2020-04-19T09:48:22.146 INFO:teuthology.orchestra.run.smithi172:> true
2020-04-19T09:48:22.164 INFO:teuthology.orchestra.run:Running command with timeout 900
```

More detail: http://qa-proxy.ceph.com/teuthology/xiubli-2020-04-19_09:09:02-fs:basic_functional-wip-lxb-volume-client-py3-2020-04-19-distro-basic-smithi/4967396/teuthology.log

Reply:
> Local test, the python2 will fail:
> 
> from ceph_volume_client import CephFSVolumeClient, VolumePath
> 
> It says couldn't find the ceph_volume_client .
> 
> As the old code logic, it will always use the python3 here.

@tchaikov 

BTW, I saw your old commit about dropping the PYTHON2 in src/pybind/

```
commit 5fc657b40dc7d27e84d4b62b5f37d9c771d74fcd
Author: Kefu Chai <kchai@redhat.com>
Date:   Wed Dec 18 17:08:21 2019 +0800

    cmake: drop WITH_PYTHON2 option
    
    * drop WITH_PYTHON2 option
    * print warning message if WITH_PYTHON3 is disabled
    * drop MGR_PYTHON_VERSION option, as we don't support use different
      python version for python binding and ceph-mgr embedded python
      interpreter anymore. as before switching to python3-only build,
      we can build python3 and python2 python bindings, and ceph-mgr
      can use either of them. but after switching to python3-only
      build, ceph-mgr has to use whatever python version used to
      build python binding.
    * move WITH_PYTHON3 option to $top_srcdir/CMakeLists.txt, as ceph-mgr
      and python binding will share this option.
    * hardware ${PYTHON_VERSION} to 3
    * hardware ${Python${PYTHON_VERSION}_VERSION_MAJOR} to 3
    * only build boost library with python3
    * s/Python_EXECUTABLE/Python3_EXECUTABLE/
    * update the build scripts and packagings accordingly
    * rename all cython${PYTHON_VERSION}_* targets to cython_*
    * update distutils_install_module() so it does not take python_version
      parameter anymore.
```
    
That's mean only the python3 will be support now, right ?
If so we need to drop the py2 test here.

Reply:
in general, yes. but i am not familiar with volume client. i am not quite sure  if we should drop python3 here as well.

Reply:
After install the ceph from the source, the ceph_volume_client package could only be find from the python3 path, as below:
```
# ls /usr/local/lib/python3.6/site-packages/__pycache__
ceph_argparse.cpython-36.pyc       easy_install.cpython-36.pyc  jsonpointer.cpython-36.pyc
ceph_volume_client.cpython-36.pyc  jsonpatch.cpython-36.pyc     portend.cpython-36.pyc
```
The old code of the this test case was buggy, it will always to set the self.py_version to python3, so we didn't hit the issue before:

```
@@ -19,7 +19,7 @@ class TestVolumeClient(CephFSTestCase):
     def setUp(self):
         CephFSTestCase.setUp(self)
         self.py_version = self.ctx.config.get('overrides', {}).\
-                          get('python3', TestVolumeClient.default_py_version)
+                          get('python', TestVolumeClient.default_py_version)
         log.info("using python version: {python_version}".format(
             python_version=self.py_version
```
There is no "python3" key in the "override" dict from:
```
 # ls qa/suites/fs/basic_functional/tasks/volume-client/py/
2.yaml 3.yaml
```
Such as:
```
# cat qa/suites/fs/basic_functional/tasks/volume-client/py/3.yaml 
overrides:
  python: python3
```

Reply:
if volume-client should by aligned with ceph regarding to python3 support, we should remove `qa/suites/fs/basic_functional/tasks/volume-client/py` facets. and always use python3.

Reply:
Okay, I will remove it. Thanks.

Reply:
and who's the consumer of volume client apart from the ones in ceph?

Reply:
It is:
`qa/workunits/fs/upgrade/volume_client`

Reply:
I fixed it together in this PR.